src/consumer/__tests__/errorRecovery.spec.js

```diff
@@ -204,7 +204,7 @@ describe('Consumer', () => {
     await waitFor(() => restartOnFailure.mock.calls.length > 0)
     expect(restartOnFailure).toHaveBeenCalledWith(error)
 
-    await expect(waitFor(() => eachMessage.mock.calls.length)).resolves.toBe(1)
+    await expect(waitFor(() => eachMessage.mock.calls.length)).resolves.toBeGreaterThanOrEqual(1)
```

Comment:
Sometimes the consumer restarts before this `waitFor` is checked, which causes `eachMessage.mock.calls` to have 2 calls instead of one.